content/blog/tune-0-1-1/index.Rmd

```diff
@@ -0,0 +1,255 @@
+
+```{r ap-irr, cache = TRUE}
+library(tidymodels)
+
+data(Chicago, package = "modeldata")
+
+# Time-series resampling
+set.seed(7898)
+data_folds <-
+  rolling_origin(
+    Chicago,
+    initial = 364 * 15,
+    assess = 7 * 4,
+    skip = 7 * 4,
+    cumulative = FALSE
+  )
+
+svm_mod <- 
+  svm_rbf(cost = tune(), rbf_sigma = tune("kernel parameter")) %>% 
+  set_mode("regression") %>% 
+  set_engine("kernlab")
+
+ctrl <- control_grid(save_pred = TRUE)
+
+set.seed(2893)
+non_regular_grid <-
+  svm_mod %>%
+  tune_grid(
+    ridership ~ Harlem + Archer_35th,
+    resamples = data_folds,
+    grid = 36,
+    control = ctrl
+  )
+```
+
+```{r irreg-plot-old, eval = FALSE, dev = 'svg', dev.args = list(bg = "transparent"), fig.align='center', fig.height=4}
+autoplot(non_regular_grid, metric = "rmse") + 
+  ggtitle("old method, irregular grid")
+```
+<img src="figure/irreg-plot-old-1.svg" title="plot of chunk irreg-plot-old" alt="plot of chunk irreg-plot-old" style="display: block; margin: auto;" />
+
+Not bad but it could be better in a few ways: 
+
+ * Both tuning parameters are generated on log scales. The data are shown above in the natural units and the data at the low end of the scale gets smashed together. 
+ 
+ * We could show its parameter label (e.g. "Radial Basis Function sigma") when no parameter ID is given. 
+ 
+What happens when a regular (i.e. factorial) grid is used? 
+
+```{r ap-reg, cache = TRUE}
+grid <- 
+  svm_mod %>% 
+  parameters() %>% 
+  grid_regular(levels = c(3, 12))
+
+grid
+
+set.seed(2893)
+regular_grid <-
+  svm_mod %>%
+  tune_grid(
+    ridership ~ Harlem + Archer_35th,
+    resamples = data_folds,
+    grid = grid,
+    control = ctrl
+  )
+```
+
+
+```{r reg-plot-old, eval = FALSE, dev = 'svg', dev.args = list(bg = "transparent"), fig.align='center', fig.height=4}
+autoplot(regular_grid, metric = "rmse") + 
+  ggtitle("old method, regular grid")
+```
+<img src="figure/reg-plot-old-1.svg" title="plot of chunk reg-plot-old" alt="plot of chunk reg-plot-old" style="display: block; margin: auto;" />
+
+This could be better since there might be a pattern in one parameter for each value of the other. 
+
+The new version of tune shows improved versions of both of these plots: 
+
+```{r ap-irr-new, dev = 'svg', dev.args = list(bg = "transparent"), fig.align='center', fig.height=4}
+autoplot(non_regular_grid, metric = "rmse") + 
+  ggtitle("new method, irregular grid")
+```
+
+This tells a completely different story than the previous version where the parameters were in their natural units. 
+
+The regular grid results are much better and tell a cleaner story: 
+
+```{r ap-reg-new, dev = 'svg', dev.args = list(bg = "transparent"), fig.align='center', fig.height=4, fig.width=5.5}
+autoplot(regular_grid, metric = "rmse") + 
+  ggtitle("new method, regular grid")
+```
+
+Extra arguments can be passed when a numeric grouping column is used and these are given to `format()`. To avoid scientific notation:
```

Comment:
```suggestion
Extra arguments can be passed when a numeric grouping column is used; these are given to `format()`. To avoid scientific notation:
```